app/src/pages/inside/settingsPage/generalTab/generalTab.jsx

```diff
@@ -0,0 +1,321 @@
+import React, { Component } from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { Input } from '../../../../components/inputs/input';
+import styles from './generalTab.scss';
+import { InputDropdown } from '../../../../components/inputs/inputDropdown';
+import { InputBigSwitcher } from '../../../../components/inputs/inputBigSwitcher';
+import { BigButton } from '../../../../components/buttons/bigButton';
+
+const cx = classNames.bind(styles);
+const cellClasses = cx('gridCell', 'centered');
+const cellClassesSubmit = cx('gridCell', 'submitButton');
+const cellClassesFormField = cx('gridCell', 'formField');
+const cellClassesColumn = cx('gridCell', 'column');
+const cellClassesEmpty = cx('gridCell', 'empty');
+
+const optionsLaunch = [
+  {
+    value: '1H',
+    label: '1 hour',
+    disabled: false,
+  },
+  {
+    value: '3H',
+    label: '3 hours',
+    disabled: false,
+  },
+  {
+    value: '6H',
+    label: '6 hours',
+    disabled: false,
+  },
+  {
+    value: '12H',
+    label: '12 hours',
+    disabled: false,
+  },
+  {
+    value: '1D',
+    label: '1 day',
+    disabled: false,
+  },
+  {
+    value: '1W',
+    label: '1 week',
+    disabled: false,
+  },
+];
+
+const options2 = [
+  {
+    value: '2W',
+    label: '2 weeks',
+    disabled: false,
+  },
+  {
+    value: '1M',
+    label: '1 month',
+    disabled: false,
+  },
+  {
+    value: '3M',
+    label: '3 months',
+    disabled: false,
+  },
+  {
+    value: '6M',
+    label: '6 months',
+    disabled: false,
+  },
+  {
+    value: 'F',
+    label: 'Forever',
+    disabled: false,
+  },
+];
+
+export class GeneralTab extends Component {
+  static defaultProps = {
+    name: 'user',
+    inactivityTimeout: '1H',
+    keepLogs: '2W',
+    keepAttachments: '2W',
+    autoAnalysis: false,
+    autoAnalysisBase: PropTypes.string,
+    submitFormData: () => {},
+  };
+
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    inactivityTimeout: PropTypes.string.isRequired,
+    keepLogs: PropTypes.string.isRequired,
+    keepAttachments: PropTypes.string.isRequired,
+    autoAnalysis: PropTypes.bool.isRequired,
+    autoAnalysisBase: PropTypes.string,
+    submitFormData: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = props;
+  }
+
+  onTimeoutChange = (e) => {
+    this.setState({
+      inactivityTimeout: e,
+    });
+  };
```

Comment:
there is a shorter version
`onTimeoutChange = (inactivityTimeout) => this.setState({ inactivityTimeout});`